### PR TITLE
DEV2-3538: Only wait for binary status after it was started

### PR DESCRIPTION
**Pull request by assapir**
PAY ATTENTION: THIS IS A BRANCH THAT IS FROM 3.7.1!!! Please don't mind unless you know what you are doing!

### CODE REVIEW

File: src/enterprise/statusBar/StatusBar.ts [hunk 1]
```diff
@@ -46,28 +40,15 @@ export class StatusBar implements Disposable {
...
   private async setServerRequired() {
-    this.item.setWarning("Please set your Tabnine server URL");
-
     if (await isHealthyServer()) {
       this.setDefaultStatus();
-      this.waitForProcess();
     } else {
+      this.item.setWarning("Please set your Tabnine server URL");
       this.item.setCommand(StatusState.SetServer);
-      this.disposables.push(
```
Comment (assapir) [inline src/enterprise/statusBar/StatusBar.ts:56]: this was removed because we are now doing it from outside

**Comment by dimacodota** [comment]
LGTM

File: src/binary/Binary.ts [hunk 2]
```diff
@@ -34,7 +34,13 @@ export default class Binary {
 
   private ready = new EventEmitter<void>();
 
+  private isReady = false;
+
   public onReady = new Promise((resolve) => {
+    if (this.isReady) {
```
[the remark this answers is not in the document]
Comment (assapir) [reply src/binary/Binary.ts:40]: I am afraid we will have a race condition where `onReady` is called after it was already fired

[the remark this answers is not in the document]
Comment (dimacodota) [reply src/binary/Binary.ts:40]: well it seems redundant, this promise executor callback will be called  once binary class is created, which always  will be before the `binary.init()` call

**Review by dimacodota — APPROVED** [review]
lgtm, see a minor comment

File: src/binary/Binary.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (assapir) [reply src/binary/Binary.ts:40]: removed

**Pull request merged**